gitlab-runner/src/run.rs

```diff
@@ -201,15 +201,21 @@ impl Run {
         interval
     }
 
-    async fn update(&self, state: JobState, cancel_token: &CancellationToken) {
+    async fn update(&self, state: JobState, cancel_token: &CancellationToken) -> Option<Duration> {
         match self
             .client
             .update_job(self.response.id, &self.response.token, state)
             .await
         {
-            Ok(_reply) => (),
-            Err(crate::client::Error::JobCancelled) => cancel_token.cancel(),
-            Err(err) => warn!("Failed to update job status: {:?}", err),
+            Ok(reply) => reply.trace_update_interval,
```

Comment:
@sjoerdsimons, what is the idea behind `trace_update_interval` and friends? It seems to be a significant amount of scaffolding for something that does not seem to be used.

Is returning it with nothing checking the returned value the right thing to do, at least for the moment?

Reply:
As mentioend else wehere, just mark that field as `#[allow(dead_code)]` for now to let it get past clippy by document that this is expected; Changing the code flow in a way that ends up shutting up clippy, but the situation doesn't change isn't really an improvement :)